src/main/java/com/cdancy/bitbucket/rest/features/SearchApi.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.bitbucket.rest.features;
+
+import com.cdancy.bitbucket.rest.annotations.Documentation;
+import com.cdancy.bitbucket.rest.binders.BindSearchRequestToPayload;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.RequestFilters;
+
+import com.cdancy.bitbucket.rest.domain.search.SearchRequest;
+import com.cdancy.bitbucket.rest.domain.search.SearchResult;
+import com.cdancy.bitbucket.rest.filters.BitbucketAuthenticationFilter;
```

Comment:
:warning: Wrong lexicographical order for 'com.cdancy.bitbucket.rest.filters.BitbucketAuthenticationFilter' import. Should be before 'org.jclouds.rest.annotations.RequestFilters'. (#QR68:35#)-CHECKSTYLE_CustomImportOrderCheck